persistence-modules/hibernate-annotations/src/main/java/com/baeldung/hibernate/customtypes/AddressType.java

```diff
@@ -92,57 +69,52 @@ public boolean equals(Object x, Object y) throws HibernateException {
     }
 
     @Override
-    public int hashCode(Object x) throws HibernateException {
+    public int hashCode(Address x) {
         return x.hashCode();
     }
 
     @Override
-    public Object nullSafeGet(ResultSet rs, String[] names, SharedSessionContractImplementor session, Object owner) throws HibernateException, SQLException {
-
+    public Address nullSafeGet(ResultSet rs, int position, SharedSessionContractImplementor session, Object owner) throws SQLException {
         Address empAdd = new Address();
-        empAdd.setAddressLine1(rs.getString(names[0]));
+        empAdd.setAddressLine1(rs.getString(position));
 
         if (rs.wasNull())
             return null;
 
-        empAdd.setAddressLine2(rs.getString(names[1]));
-        empAdd.setCity(rs.getString(names[2]));
-        empAdd.setCountry(rs.getString(names[3]));
-        empAdd.setZipCode(rs.getInt(names[4]));
+        empAdd.setAddressLine2(rs.getString(position));
```

Comment:
I just migrated to the new schema and the only parameter that I was able to assign was an integer instead of array. I will further investigate your question.

Reply:
I looked over this class and it seems that the `nullSafeGet` method is not called event if one of the parameter is null (when do the save). Even if I return null from `nullSafeGet` nothing failed.
Do you have any idea or any suggestion?

Reply:
Somehow I respond to this question. I tried to set values on the `empAdd` null and store it. I expected that the after I query that record `nullSafeGet` will be triggered and call that method, this is not happened. If I make a simple a simple query to fetch that record the values are fetched correctly. 
e.g: 
`session.createQuery("FROM OfficeEmployee OE WHERE OE.empAddress.zipcode = :pinCode", OfficeEmployee.class)` 
If I ran this query the values on the `empAddress` are correctly, that is why I asked for your help/ further suggestion.